src/test/java/com/github/redouane59/twitter/nrt/ITwitterClientV2Test.java

```diff
@@ -32,20 +35,44 @@
 public class ITwitterClientV2Test {
 
   private static TwitterClient twitterClient;
-  private        String        userId = "1307302673318895621";
+  private String userId = "1307302673318895621";
 
   @BeforeAll
   public static void init() {
     twitterClient = new TwitterClient();
+    twitterClient.setAPIListener(new IAPIEventListener() {
```

Comment:
@hypr2771 any feedback on this maybe ? Before no call to setter was needed and the signature of the method was ` void startFilteredStream(Consumer<Tweet> consumer)` . No the signature doesn't have any argument.

Reply:
Clearly, it's a major change.
When i added an existing stream, there was a "duplicate rules error" with httpcode 201 as a feedback from the TwitterAPI, but an IllegalArgumentException. Having a global API Exception Handler is a must have in my point of view ( for logging & debugging purpose).

Reply:
I like the idea but I would prefer to keep both solutions, in order to have one simple way to instantiate it quickly with little code, and a second way to fully managed everything. What do you think ?

Reply:
I agree with @redouane59 , that it will be nice to keep old approach as well.

Reply:
See the answer below.